### ISSUE TITLE
Sprint 1.39.0 1/2 - July 25 to Aug 5

### ISSUE THREAD

**Issue by marcushyett-ph** [PostHog/posthog#10893]
# Global Sprint Planning

## Retro: Status of Outcomes from Previous Sprint

1. Kick off the persons on events async migration on Cloud and get a self-hosted team to run it (@yakkomajuri) (Status: going to roll over - the migration is effectively ready but we should run it on a weekend for minimal disruption, and we were slowed down by Karl catching COVID and Yakko helping customers while Tiina was off. Great progress on important side quests though, like historical exports and the buffer 3.0)
    - Why: Unlock high scale querying
2. (P0) PGBouncer clean shutdown and connection drain on helm chart upgrade / deployments, (P0) 5% -> 50% -> 100% of events on EKS by end of sprint (Ambitious), (P1) EU deployment planning, (P1) First part data collection (proxy, DNS, something) @fuziontech (Status: updates below!)
3. Complete S3 Backed Session Recordings (@rcmarron) (Status:❓Update please)
   - Why: Longer session retention
4. Client Library support for Feature flags, sessions & groups (Android, iOS) @alexkim205 (Status: Android and IOS are going to be done by EOW)
   - Why: Increase surface area for these features in most used libraries
5. Build Expo compatible reactive native library @benjackwhite (Status: 95% there, just waiting on npm building /publishing/testing)
   - Why: 1/3 of Web events come from react native, we don't support expo
6. Onboarding prompts @kappa90 (Status: 80% there, releasing a PR today, will need some rounds of reviews)
   - Why: Increasing activation beyond the ingestion part of onboarding

## Retro: What can we do better next sprint?

1.
2.
3.
4.
5.

## Support hero this sprint

Week 1: 
Week 2:

## Plan: Proposed Goals for Next Sprint 

Each goal should have a single owner. Owner can only be an engineer.

1. Kick off the persons on events async migration on Cloud and get a self-hosted team to run it @macobo
    - Why: Unlock high scale querying
2. Targeting a 'no rush' 100% - If we identify opportunities to improve the EKS stack we will before switching over @hazzadous
    - Why: Moving to unified infrastructure
4. Continue supporting our paying on prem customers (big companies & co) @fuziontech
    - Why: Unblock key customers
6. EU Deployment plan (Project docs) @guidoiaquinti 
    - Why: Unblock EU based customers
8. S3 Session Recording @rcmarron
   - Why: Increase retention to enable more use of session recordings
1. Feature flag @neilkakkar Local evaluation of feature flags on server libraries (Python,Go,Ruby,Node,PHP): https://github.com/PostHog/posthog/issues/7115
   - Why: Faster evaluation to unblock key clients
1. Data Management Finalization + Recordings data management https://github.com/PostHog/posthog/issues/10892 @alexkim205
   - Why: Finishing off data management changes and customer issues from last week
1.  We will have agreed on and tested the tooling and architecture of the design system. @benjackwhite 
   - Why: PostHog engineers are slowed down with our legacy frontend practices

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
Team ___

## Retro

<!-- Talk about what went well, what didn't go well and any actions to improve next time -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## Planning

<!-- Each item should have a single owner. Owner can only be an engineer. -->

### High priority

-

### Bugs

-

### Low priority / side quests

-

```

**Comment by yakkomajuri** [comment]
# Team Ingestion

## Goal

Kick off the persons on events async migration on Cloud and get a self-hosted team to run it @macobo 
Why: Unlock high scale querying

**Notes**

-Yakko will be off for the whole sprint

## Retro

- @tiina303:
    - worried that working on data integrity "side quests" left @yakkomajuri as a lone wolf but the work was important to do
    - enjoy working on data integrity
    - vacation was nice (130kg deadlift PB!!)
- @yakkomajuri:
    - spent 1/4 of time on hiring
    - otherwise felt very productive working on "side quests": buffer, historical exports, plugin issues, metrics, etc
        - this was important work that was being neglected
    - not super happy that we didn't achieve the main goal, but with Karl out and customers having issues I think we made the right calls on prioritization

**Comment by fuziontech** [comment]
Team Platform Infrastructure

## Retro

@hazzadous - Didn't do any infra - just messed around with kafka stuff. Session recordings. Wants to come to California to hang out with @rcmarron 

@guidoiaquinti - Busy - lone wolfing it - Great hanging out with @hazzadous in Barcelona.

@fuziontech - Busy - Issues with client library for DE use cases. There have been legitimate issues scaling the cluster. A bunch of support.

## Hang over items from previous sprint

(P0) 
- PGBouncer clean shutdown and connection drain on helm chart upgrade / deployments 5% -> 50% -> 100% of events on EKS by end of sprint (Ambitious)
  - The PG Bouncer is sorted. Metrics for PG Bouncer which is great.
(P1)
- EU deployment planning
  - In @guidoiaquinti head, will write down soon
- First part data collection (proxy, DNS, something)
  - @timgl made some progress using CloudFlare workers
  - Need to test this to confirm

## Planning

(P0)
- Targeting a 'no rush' 100% - If we identify opportunities to improve the EKS stack we will before switching over @hazzadous 
- Continue supporting our paying on prem customers (big companies & co) @fuziontech 

(P1)
- EU Deployment plan (Project docs) @guidoiaquinti

**Comment by mariusandra** [comment]
Team App East

## Retro

TODO

## Planning

- Objective: **Make it easy to build new features quickly that fit in with our design system.**
- Key Results:  50% reduction in custom and inline CSS styles. Exact metric TBD.
- Rationale: PostHog engineers are slowed down with our legacy frontend practices.
- Goal: **We will have agreed on and tested the tooling and architecture of the design system.**


Some subgoals, will be broken down in the next meeting.

- Initial scope: 
  - Clean up:
    - Form elements
    - Menu items
    - Buttons
  - Consider linter rules for components and props we want to not use anymore
  - Investigate to use Tailwind or not

**Comment by EDsCODE** [comment]
**Priorities:**
- S3 Session Recording @rcmarron 
- Feature flag @neilkakkar 
     - Local evaluation of feature flags on server libraries
          - Python
          - Go
          - Ruby
          - Node
          - PHP
- Data Management Finalization + Recordings data management @alexkim205 
    - Focus on what’s important for getting data management pushed
    - https://github.com/PostHog/posthog/issues/10892

**Comment by weyert** [comment]
@EDsCODE @neilkakkar The local evaluation of feature flags on server library sounds interested. Do you have a ticket or more details about this task?

**Comment by neilkakkar** [comment]
Yep: https://github.com/PostHog/posthog/issues/7115
